packages/rpc/json2/server.go

```diff
@@ -0,0 +1,259 @@
+// Copyright 2020 ChainSafe Systems (ON) Corp.
+// This file is part of gossamer.
+//
+// The gossamer library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+
+package json2
```

Comment:
what is the file used for?

Reply:
This is the custom JSON to struct decoder we are using to parse unstructured JSON params. 
I have taken the code for `CodecRequest` from this https://github.com/gorilla/rpc/blob/master/v2/json2/server.go and fixed it to work.